### DISCUSSION TITLE
autoCreateTopicEnable=true is not work

### DISCUSSION THREAD

**Discussion by Yknight-dev** [apache/rocketmq#7378] [General]
**rocketMq version**:  rocketmq-all-5.1.3-bin-release
**client version** : rocketmq-client-java 5.0.5
### producer code :
             
    public static void main(String[] args) throws ClientException {
        // 接入点地址，需要设置成Proxy的地址和端口列表，一般是xxx:8081;xxx:8081。
        String endpoint = "localhost:8081";
        // 消息发送的目标Topic名称，需要提前创建。
        String topic = "orderTopic4";
        ClientServiceProvider provider = ClientServiceProvider.loadService();
        ClientConfigurationBuilder builder = ClientConfiguration.newBuilder().setEndpoints(endpoint);
        ClientConfiguration configuration = builder.build();
        // 初始化Producer时需要设置通信配置以及预绑定的Topic。
        Producer producer = provider.newProducerBuilder()
                .setTopics(topic)
                .setClientConfiguration(configuration)
                .build();

        // Define your message body.
        byte[] body = "This is a FIFO message for Apache RocketMQ".getBytes(StandardCharsets.UTF_8);
        String tag = "yourMessageTagA";
        final Message message = provider.newMessageBuilder()
                // Set topic for the current message.
                .setTopic(topic)
                // Message secondary classifier of message besides topic.
                .setTag(tag)
                // Key(s) of the message, another way to mark message besides message id.
                .setKeys("yourMessageKey-1ff69ada8e0e")
                // Message group decides the message delivery order.
                .setMessageGroup("yourMessageGroup0")
                .setBody(body)
                .build();
        try {
            final SendReceipt sendReceipt = producer.send(message);
            log.info("Send message successfully, messageId={}", sendReceipt.getMessageId());
        } catch (Throwable t) {
            log.error("Failed to send message", t);
        }
        // Close the producer when you don't need it anymore.
        // You could close it manually or add this into the JVM shutdown hook.
        // producer.shutdown();
    }

**question:**   the topic "orderTopic4" not exist,it report a bug with  `DESC: No topic route info in name server for the topic: orderTopic4`
I think i open `the autoCreateTopicEnable=true` but it doesn't create the topic ,it doesn't work

**Comment by RongtongJin** [comment]
在RocketMQ 5.0，该参数只有直连broker才生效，连接proxy不生效

**Reply by Yknight-dev** [reply]
文档中哪里有，请给我一个链接

**Reply by joeCarf** [reply]
这是否是一个待完成的功能呢？或是有意设计成这样的吗。如果是功能的话，我可以尝试实现一下

**Reply by onejimmyboy** [reply]
我忘记在那个地方看到过，之前也有社区提出过这个问题，当时有人回复的是不建议支持，在实际生成环境中，都是建议关闭这个配置的。

**Reply by onejimmyboy** [reply]
感觉RocketMQ到5.0后，现在的proxy模式，开发文档上面有很多欠缺的，并且和dashboard有一些没有做好适配，dashboard又好长时间没有更新了。心塞啊

**Reply by Yknight-dev** [reply]
好的，感谢。我直接连接nameServer就可以自动创建主题了。开发环境是不建议的，但是有点强迫症所以想知道原因
